pkg/csi/snapshot_restore.go

```
@@ -261,44 +278,50 @@ func (s *snapshotRestoreSteps) RestoreApplication(ctx context.Context, args *typ
 	if err != nil {
 		return nil, pvc, errors.Wrap(err, "Failed to create restored Pod")
 	}
+
+	if err = s.createAppOps.WaitForPVCReady(ctx, args.Namespace, pvc.Name); err != nil {
+		return pod, pvc, errors.Wrap(err, "PVC failed to become ready")
+	}
+
 	if err = s.createAppOps.WaitForPodReady(ctx, args.Namespace, pod.Name); err != nil {
 		return pod, pvc, errors.Wrap(err, "Pod failed to become ready")
 	}
 	return pod, pvc, nil
 }
 
-func (s *snapshotRestoreSteps) Cleanup(ctx context.Context, results *types.CSISnapshotRestoreResults) {
+func (s *snapshotRestoreSteps) Cleanup(results *types.CSISnapshotRestoreResults) {
```

Why are you changing the function signature by removing the context?

---

The context is being used.

---

How does this help "Fail fast"

---

It doesn't help "fail fast"
It refers to the "and cleanup regardless". It also prevents https://github.com/kastenhq/kubestr/issues/102

---

When PVC/Pod hangs the passed ctx gets DeadlineExceeded  and cleanup never happens. This avoids that issues

---

unless someone outside this repo might be using this API directly (not sure how likely)...then could basically keep the passed context but just ignore it although it looks kinder weird

---

I changed the timeout to 2 mins for the PVC and then the Pod being ready (so 4 mins total worst case). There's no need to have a timeout on kubectl commands; the hanging is all in the PVCs/Pods reaching ready state. The cleanup here just requests deletion, it doesn't wait around for deletion to complete (unlike I do in k10 to allow 1 retry of creation for a specific scenario [`'resource.sizeGb': '1'. Disk size cannot be smaller than 200 GB`])

---

Yes, you are right. Delete does return immediately, even though the underlying resources are stuck in the `Terminating` state, which is a separate problem.